extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -257,6 +273,45 @@ export class SubscriptionService {
     return this;
   }
 
+  /**
+   * Delegates authentication to viewer
+   */
+  delegateAuthToViewer_() {
+    const serviceIds = ['local'];
+    const currentProductId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getProductId(),
+        'Product id is null'
+    ));
+    const publicationId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getPublicationId(),
+        'Publication id is null'
+    ));
+    const origin = getWinOrigin(this.ampdoc_.win);
+    this.platformStore_ = new PlatformStore(serviceIds);
+    this.viewer_.sendMessageAwaitResponse('auth', dict({
+      'publicationId': publicationId,
+      'productId': currentProductId,
+      'origin': origin,
+    })).then(entitlementData => {
+      const authData = (entitlementData || {})['authorization'];
+      dev().assert(authData, 'authorization is not defined');
+      const decodedData = this.jwtHelper_.decode(authData);
+      if (decodedData['aud'] != origin) {
+        user().error(TAG, 'Audience does not match');
```

Comment:
This is just log. It will continue executing down the path. If you want an exception, it's either `user().assert(decodedData['aud'] == origin, 'bad audience')` or `throw user().createError(...)`.